packages/ui/src/components/atoms/Dropdown.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useEffect, useRef, useState } from 'react';
+import styled from 'styled-components';
+
+import { Image } from './Image';
+import { Input } from './Input';
+
+import { searchIcon, triangleInverseIcon, walletIcon } from '../../assets';
+import { theme } from '../../themes/theme.styled';
+import { LeanBoxProps, DropDownListItem } from '../molecules';
+
+interface DropdownProps {
+  items: LeanBoxProps[];
+  changeColorWhite?: boolean;
+  searchText: string;
+  placeholderText: string;
+  selectedItem: string | undefined;
+  onChange: (selectedItemId: string | undefined) => void;
+  disabled?: boolean;
+}
+
+const List = styled.ul<{ disabled?: boolean }>`
+  position: absolute;
+  top: 100%;
+  right: 0;
+  width: 100%;
+  list-style: none;
+  border-radius: 8px;
+  box-shadow: 4px 4px 32px 4px ${theme.palette.shadow.dropdown};
+  padding: 16px 0px 16px 0px;
+  z-index: 10;
+  background-color: ${theme.palette.background.input};
+  &:hover {
+    cursor: ${props => (!props.disabled ? 'pointer' : 'default')};
+  }
+`;
+
+const ListItem = styled.li`
+  background-color: ${theme.palette.border.list};
+`;
+
+const Container = styled.div<{ isOpen: boolean; disabled?: boolean }>`
+  position: relative;
+  width: 100%;
+  border-radius: 8px;
+  background-color: ${theme.palette.border.dropdown};
+  border: 2px solid
+    ${props =>
+      props.isOpen ? theme.palette.border.gold : theme.palette.border.dropdown};
+  &:hover {
+    border: 2px solid
+      ${props =>
+        !props.disabled
+          ? theme.palette.border.gold
+          : theme.palette.border.dropdown};
+  }
+  input {
+    padding-right: 30px;
+    cursor: ${props => (props.disabled ? 'default' : 'pointer')};
+  }
+`;
+
+const IconContainer = styled.div`
+  position: absolute;
+  right: 24px;
+  top: 60%;
+  transform: translateY(-50%);
+`;
+
+export const Dropdown: React.FC<DropdownProps> = ({
+  items,
+  changeColorWhite,
+  searchText,
+  placeholderText,
+  selectedItem = undefined,
+  onChange,
+  disabled = false,
+}) => {
+  const [search, setSearch] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+  const [checkedStates, setCheckedStates] = React.useState<
+    Record<string, boolean>
+  >({});
```

Comment:
What's the difference between `checkedStates` and `selectedItem`? I think any one of them is sufficiant.

Reply:
removed